lib/node_modules/@stdlib/random/base/tinymt32/lib/factory.js

```diff
@@ -0,0 +1,556 @@
+var MAX_NORMALIZED = FLOAT64_MAX_SAFE_INTEGER * FLOAT64_NORMALIZATION_CONSTANT;
+
+// Define the state array schema version:
+var STATE_ARRAY_VERSION = 1; // NOTE: anytime the state array schema changes, this value should be incremented!!!
+
+// Define the number of sections in the state array:
+var NUM_STATE_SECTIONS = 2; // state, seed
+
+// Define the index offset of the "state" section in the state array:
+var STATE_SECTION_OFFSET = 2; // | version | num_sections | state_length | ...state | state_index | seed_length | ...seed |
+
+// Define the index offset of the seed section in the state array:
+var SEED_SECTION_OFFSET = N + 4; // | version | num_sections | state_length | ...state | state_index | seed_length | ...seed |
+
+// Define the length of the "fixed" length portion of the state array:
+var STATE_FIXED_LENGTH = N + 5; // 1 (version) + 1 (num_sections) + 1 (state_length) + N (state) + 1 (state_index) + 1 (seed_length)
+
+
+// TinyMT32 constants
+var MAT1 = 0x8F7011EE;
+var MAT2 = 0xFC78FF1F;
+var TMAT = 0x3793FDFF;
+
+// TinyMT32 shift constants
+var SH0 = 1;
+var SH1 = 10;
+var SH8 = 8;
+var MASK = 0x7FFFFFFF;
+
+// Initialization constants
+var MIN_LOOP = 8;
+var PRE_LOOP = 8;
+var KNUTH_MULTIPLIER = 1812433253 >>> 0; // asm type annotation
+
+// FUNCTIONS //
+
+/**
+* Verifies state array integrity.
+*
+* @private
+* @param {Uint32Array} state - state array
+* @param {boolean} FLG - flag indicating whether the state array was provided as an option (true) or an argument (false)
+* @returns {(Error|null)} an error or `null`
+*/
+function verifyState( state, FLG ) {
+    var s1;
+    if ( FLG ) {
+        s1 = 'option';
+    } else {
+        s1 = 'argument';
+    }
+    if ( state.length < STATE_FIXED_LENGTH+1 ) {
+        return new RangeError( format( 'invalid %s. State array has insufficient length.', s1 ) );
+    }
+    if ( state[ 0 ] !== STATE_ARRAY_VERSION ) {
+        return new RangeError( format( 'invalid %s. State array has an incompatible schema version. Expected: `%s`. Actual: `%s`.', s1, STATE_ARRAY_VERSION, state[ 0 ] ) );
+    }
+    if ( state[ 1 ] !== NUM_STATE_SECTIONS ) {
+        return new RangeError( format( 'invalid %s. State array has an incompatible number of sections. Expected: `%s`. Actual: `%s`.', s1, NUM_STATE_SECTIONS, state[ 1 ] ) );
+    }
+    if ( state[ STATE_SECTION_OFFSET ] !== N ) {
+        return new RangeError( format( 'invalid %s. State array has an incompatible state length. Expected: `%u`. Actual: `%u`.', s1, N, state[ STATE_SECTION_OFFSET ] ) );
+    }
+    if ( state[ SEED_SECTION_OFFSET ] !== state.length-STATE_FIXED_LENGTH ) {
+        return new RangeError( format( 'invalid %s. State array length is incompatible with seed section length. Expected: `%u`. Actual: `%u`.', s1, state.length-STATE_FIXED_LENGTH, state[ SEED_SECTION_OFFSET ] ) );
+    }
+    return null;
+}
+
+function multiplyUnsigned(a, b) {
```

Comment:
Not clear why you are not using `number/uint32/base/mul`, rather than including directly here.